integration_test/upgrade_test.go

```diff
@@ -50,6 +51,38 @@ func TestKrewUpgrade(t *testing.T) {
 	}
 }
 
+func TestKrewUpgradePluginsFromCustomIndex(t *testing.T) {
+	skipShort(t)
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.WithEnv(constants.EnableMultiIndexSwitch, 1).WithIndex()
+	test.Krew("index", "add", "foo", test.TempDir().Path("index/"+constants.DefaultIndexName)).RunOrFail()
+	test.Krew("install", "foo/"+validPlugin).RunOrFail()
+
+	receipt := environment.NewPaths(test.Root()).PluginInstallReceiptPath(validPlugin)
+	modifyManifestVersion(t, receipt, "v0.0.0")
+	out := string(test.Krew("upgrade").RunOrFailOutput())
+	if !strings.Contains(out, "Upgrading plugin: foo/"+validPlugin) {
+		t.Errorf("expected plugin foo/%s to be upgraded", validPlugin)
+	}
+
+	modifyManifestVersion(t, receipt, "v0.0.0")
+	out = string(test.Krew("upgrade", validPlugin).RunOrFailOutput())
+	if !strings.Contains(out, "Upgrading plugin: foo/"+validPlugin) {
+		t.Errorf("expected plugin foo/%s to be upgraded", validPlugin)
+	}
+
+	b, err := test.Krew("upgrade", "foo/"+validPlugin).Run()
+	if err == nil {
+		t.Error("expected error when using canonical name with upgrade")
+	}
+	if !strings.Contains(string(b), "INDEX/PLUGIN") {
+		t.Error("expected warning about using canonical name to be in output")
+	}
```

Comment:
this is dope but do you mind extracting into a test named like `TestKrewUpgrade_CannotUseIndexSyntax`

You don't have to (1) add index (2) install plugin (3) specify foo/validPlugin (even x/y works) you should be getting the same error.